Source/WebCore/animation/StyleOriginatedTimelinesController.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2025 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "ScrollAxis.h"
+#include "Styleable.h"
+#include "TimelineScope.h"
+#include <wtf/CheckedRef.h>
+#include <wtf/WeakHashSet.h>
+#include <wtf/text/AtomStringHash.h>
+
+namespace WebCore {
+
+class CSSAnimation;
+class ScrollTimeline;
+class WebAnimation;
+
+struct ViewTimelineInsets;
+
+struct TimelineMapAttachOperation {
+    WeakStyleable element;
+    AtomString name;
+    Ref<CSSAnimation> animation;
+};
+
+DECLARE_ALLOCATOR_WITH_HEAP_IDENTIFIER(StyleOriginatedTimelinesController);
+class StyleOriginatedTimelinesController final : public CanMakeCheckedPtr<StyleOriginatedTimelinesController> {
```

Comment:
This would be a great place to add a brief comment to explain what a "style-originated" timeline is. Does it include CSS Animation timelines?

Reply:
Good point! A style-originated timeline is a timeline that is assigned to a CSS Animation via the `animation-timeline` property and which may also involve the `scroll-timeline-name`, `view-timeline-name` and `timeline-scope` properties. I'll make sure to add this as a comment.